crowbar_framework/app/models/api/node.rb

```diff
@@ -72,7 +69,26 @@ def os_upgrade
         Rails.logger.error("Executing of os upgrade script has failed on node #{@node.name}.")
         return false
       end
-      true
+
+      begin
```

Comment:
Not sure I understand. A method called `#wait_for_script_to_finish` would just do that, not any extra stuff like updating upgrade status.  Any upgrade-specific stuff would be done in a separate method in one of the upgrade-specific classes. And it might make sense to build another upgrade-specific method which wraps both methods, e.g. `#upgrade_step_via_script` which calls `#wait_for_script_to_finish` and then `#end_step`.